Hazel/src/Hazel/Scene/SceneSerializer.cpp

```diff
@@ -82,70 +82,86 @@ namespace Hazel {
 
-			auto& tc = entity.GetComponent<TransformComponent>();
-			out << YAML::Key << "Translation" << YAML::Value << tc.Translation;
-			out << YAML::Key << "Rotation" << YAML::Value << tc.Rotation;
-			out << YAML::Key << "Scale" << YAML::Value << tc.Scale;
+		out << YAML::EndMap; // TransformComponent
+	}
 
-			out << YAML::EndMap; // TransformComponent
-		}
+	template<>
+	static void SerializeComponent(YAML::Emitter& out, const CameraComponent& cameraComponent)
+	{
+		out << YAML::Key << "CameraComponent";
+		out << YAML::BeginMap; // CameraComponent
+
+		auto& camera = cameraComponent.Camera;
+
+		out << YAML::Key << "Camera" << YAML::Value;
+		out << YAML::BeginMap; // Camera
+		out << YAML::Key << "ProjectionType" << YAML::Value << (int)camera.GetProjectionType();
+		out << YAML::Key << "PerspectiveFOV" << YAML::Value << camera.GetPerspectiveVerticalFOV();
+		out << YAML::Key << "PerspectiveNear" << YAML::Value << camera.GetPerspectiveNearClip();
+		out << YAML::Key << "PerspectiveFar" << YAML::Value << camera.GetPerspectiveFarClip();
+		out << YAML::Key << "OrthographicSize" << YAML::Value << camera.GetOrthographicSize();
+		out << YAML::Key << "OrthographicNear" << YAML::Value << camera.GetOrthographicNearClip();
+		out << YAML::Key << "OrthographicFar" << YAML::Value << camera.GetOrthographicFarClip();
+		out << YAML::EndMap; // Camera
+
+		out << YAML::Key << "Primary" << YAML::Value << cameraComponent.Primary;
+		out << YAML::Key << "FixedAspectRatio" << YAML::Value << cameraComponent.FixedAspectRatio;
+
+		out << YAML::EndMap; // CameraComponent
+	}
 
-		if (entity.HasComponent<CameraComponent>())
-		{
-			out << YAML::Key << "CameraComponent";
-			out << YAML::BeginMap; // CameraComponent
-
-			auto& cameraComponent = entity.GetComponent<CameraComponent>();
-			auto& camera = cameraComponent.Camera;
-
-			out << YAML::Key << "Camera" << YAML::Value;
-			out << YAML::BeginMap; // Camera
-			out << YAML::Key << "ProjectionType" << YAML::Value << (int)camera.GetProjectionType();
-			out << YAML::Key << "PerspectiveFOV" << YAML::Value << camera.GetPerspectiveVerticalFOV();
-			out << YAML::Key << "PerspectiveNear" << YAML::Value << camera.GetPerspectiveNearClip();
-			out << YAML::Key << "PerspectiveFar" << YAML::Value << camera.GetPerspectiveFarClip();
-			out << YAML::Key << "OrthographicSize" << YAML::Value << camera.GetOrthographicSize();
-			out << YAML::Key << "OrthographicNear" << YAML::Value << camera.GetOrthographicNearClip();
-			out << YAML::Key << "OrthographicFar" << YAML::Value << camera.GetOrthographicFarClip();
-			out << YAML::EndMap; // Camera
-
-			out << YAML::Key << "Primary" << YAML::Value << cameraComponent.Primary;
-			out << YAML::Key << "FixedAspectRatio" << YAML::Value << cameraComponent.FixedAspectRatio;
-
-			out << YAML::EndMap; // CameraComponent
-		}
+	template<>
+	static void SerializeComponent(
+		YAML::Emitter& out, const SpriteRendererComponent& spriteRendererComponent)
```

Comment:
I know the line looks long, cherno uses abriviations loke `src` for the spriteRenderComponent name, at least it should remain on a single line

```suggestion
	static void SerializeComponent(YAML::Emitter& out, const SpriteRendererComponent& spriteRendererComponent)
```

If you want you can rename them all to their abbriviation, i'm not gonna make suggestions for all of them, just for this one, but if you merge it, at least do it for all of them
```suggestion
	static void SerializeComponent(YAML::Emitter& out, const SpriteRendererComponent& src)
```
And don't forget to change it inside the function as well, obviously...

Reply:
Thanks for your feedback--these are all great points. I made this commit because I left some critical comments on the video and felt obligated to put in some work instead of just complaining. I didn't know you guys would be seriously reviewing it, so in light of that I'll retract for now and come back with a higher effort contribution. Cheers

Reply:
Any community addition is always appreciated, and there is not one way of doing something, cherno does go out of his way when the community asks something specific even, on the other hand he praises when we do something good, and he gives feedbacks if he thinks it could be better...